src/core/Global.ttslua

```diff
@@ -1035,3 +1039,92 @@ function titleSplash(scenarioName)
     soundCubeApi.playSoundByName("Deep Bell")
   end
 end
+
+---------------------------------------------------------
+-- Update notification related functionality
+---------------------------------------------------------
+
+-- grabs the latest mod version and release notes from GitHub (called onLoad())
+function getModVersion()
+  WebRequest.get(SOURCE_REPO .. '/modversion.json', compareVersion)
+end
+
+-- compares the modversion with GitHub and possibly shows the update notification
+function compareVersion(request)
+  if request.is_error then
+    log(request.error)
+    return
+  end
+
+  -- global variable to make it accessible for other functions
+  modMeta = JSON.decode(request.text)
+
+  -- stop here if on latest version or "don't show again" was clicked
+  if MOD_VERSION == modMeta["latestVersion"] then return end
+  
+  if dontNotify[modMeta["latestVersion"]] then return end
+
+  -- end here if no release notes are found
+  if modMeta["releaseNotes"][modMeta["latestVersion"]] == nil then
```

Comment:
The releaseNotes node seems unnecessary.  We shouldn't have more than a single version recorded there, so Highlights can be at the same level as latestVersion.  I think Details can be removed as well, since it's not used?

Reply:
I intend this file to be used for the whole release notes in the future. E.g. by being able to select a version in a drop-down menu and then seeing the notes for that version (which "Details" is for)

Reply:
Fair on Details, but it still shouldn't need to be a table, right?  Unless we're going to have multiple versions (and I don't think there's a need to), that layer is redundant.  Top-level fields can be latestVersion (or even just version), highlights, and details.

Reply:
Why wouldn't we have multiple entries? If we don't, you can only view the release notes of you are on the newest version.